### ISSUE TITLE
Error, AppendList: <list2> must be a small list... in membership test for matrix groups

### ISSUE THREAD

**Issue by dimpase** [gap-system/gap#826]
The following GAP code

```
i:=E(4);
S:=((1+i)/Sqrt(2))*DiagonalMat([i,i,1,1]);
T:=((1+i)/2)*[[-i,0,0,i],[0,1,1,0],[1,0,0,1],[0,-i,i,0]];
kT1:=KroneckerProduct(T,IdentityMat(2));
kT2:=KroneckerProduct(IdentityMat(2),T);
g:=Group(kT1,kT2);
Print(Order(g),"\n");
KroneckerProduct(S,IdentityMat(2)) in g;
```

bails out with

```
...
gap> g:=Group(kT1,kT2);
<matrix group with 2 generators>
gap> Print(Order(g),"\n");
371589120
gap> KroneckerProduct(S,IdentityMat(2)) in g;
Error, AppendList: <list2> must be a small list (not a boolean) in
  Append( resrow, Coefficients( B, entry * b ) ); at /home/dima/software/gap/gap4r8/lib/matrix.gi:3082 called from 
BlownUpMat( Basis( iso ), mat ) at /home/dima/software/gap/gap4r8/lib/grpmat.gi:993 called from
ImagesRepresentative( nice, elm ) at /home/dima/software/gap/gap4r8/lib/grpnice.gi:224 called from
<function "unknown">( <arguments> )
 called from read-eval loop at line 8 of *stdin*
you can replace <list2> via 'return <list2>;'
brk> 
```

I guess it has to do with `371589120>2^28` - the latter the the maximal size of a small list.
The result should be `false` in fact; this can be checked by computing the order of the group obtained from `g` by adding this extra element to the generating set, and computing the order (twice as as big as `|g|`).
Also note that removing the call to `Order(g)` leads to the test running out of RAM (probably something really slow and memory hungry is tried, like generating all the group elements?).

This is on the standard GAP 4.8.3 on x86_64:

```
 ┌───────┐   GAP 4.8.3, 19-Mar-2016, build of 2016-04-03 22:53:57 (BST)
 │  GAP  │   http://www.gap-system.org
 └───────┘   Architecture: x86_64-pc-linux-gnu-gcc-default64
 Libs used:  gmp, readline
 Loading the library and packages ...
 Components: trans 1.0, prim 2.1, small* 1.0, id* 1.0
 Packages:   AClib 1.2, Alnuth 3.0.0, AtlasRep 1.5.0, AutPGrp 1.6, CRISP 1.4.3, Cryst 4.1.12, 
             CrystCat 1.1.6, CTblLib 1.2.2, FactInt 1.5.3, FGA 1.3.1, GAPDoc 1.5.1, IO 4.4.6, 
             IRREDSOL 1.2.4, LAGUNA 3.7.0, Polenta 1.3.6, Polycyclic 2.11, RadiRoot 2.7, 
             ResClasses 4.4.2, Sophus 1.23, SpinSym 1.5, TomLib 1.2.5, Utils 0.39
```

I also checked that on 4.6.5 one gets the same error.

**Comment by markuspf** [comment]
This has nothing to do with the group size being greater than 2^28, it's because computing coefficients fails, and Append doesn't work for a list and `fail`. Rather related: #827.

This can be discovered by trying out what GAP is trying to do in the break loop:
```
brk> resrow;
[  ]
brk> Coefficients(B, entry * b);
fail
brk> entry;
E(8)^3
brk> b;
1
brk> B;
CanonicalBasis( GaussianRationals )
brk> Append(resrow, Coefficients(B, entry * b);
...
brk_2>
```

**Comment by markuspf** [comment]
It is also fairly well-known that currently matrix groups are not treated optimally. If you want a result for your example you can do the following:

```
gap> iso := IsomorphismPermGroup(g); # This will take a couple of seconds
<action isomorphism>
gap> Size(Range(iso));  # This is then instant
371589120
gap> Size(Group(GeneratorsOfGroup(Range(iso)))); # This is also almost instant, just to check I am not cheating
gap> ImagesRepresentative(iso, KroneckerProduct(S, IdentityMat(2)));
fail
gap> ImagesRepresentative(iso, KroneckerProduct(T, IdentityMat(2)));
(...) # This spews out a rather long permutation
```

I'll take a look at where the membership test for matrix group could do with a fix.

**Comment by dimpase** [comment]
Oh I see; this test should fail just because `S` (and thus the matrix we test for membership) does not lie in CF(4) (aka GaussianRationals), whereas the group does. I did not notice that. 

Should be easy to fix (although it's baffling that similar tests for smaller examples work just fine; must be a different codepath...).

**Comment by fingolfin** [comment]
A simplified version of the original test case runs much quicker and produces the same result:
```gap
i:=E(4);
S:=((1+i)/Sqrt(2))*DiagonalMat([i,i,1,1]);
T:=((1+i)/2)*[[-i,0,0,i],[0,1,1,0],[1,0,0,1],[0,-i,i,0]];
g:=Group(T);
Print(Order(g),"\n");
S in g;
```

The problem is in the end this method:
```gap
InstallMethod( ImagesRepresentative,
    "for a blow up isomorphism, and a matrix in the source",
    FamSourceEqFamElm,
    [ IsBlowUpIsomorphism, IsMatrix ],
    function( iso, mat )
    return BlownUpMat( Basis( iso ), mat );
    end );
```
This calls `BlownUpMat`, whose documentation states:
> Let `B` be a basis of a field extension `F / K`, and `mat` a matrix whose entries are all in `F`.  (This is not checked.)

I submitted a fix in PR #1488
